oceannavigator/frontend/src/components/PlotImage.jsx

```diff
@@ -40,26 +40,16 @@ export default class PlotImage extends React.PureComponent {
   }
 
   generateScript(language) {
+      const query = encodeURIComponent(stringify(this.generateQuery(this.props.query)));
       if (language == "pythonPlot") {
-        var url = stringify(this.generateQuery(this.props.query));
-        url = window.location.origin + "/api/v1.0/generatescript/" + url + "/python/" + "PLOT/";
+        var url = window.location.origin + "/api/v1.0/generatescript/?query=" + query + "&lang=python" + "&scriptType=PLOT";
       } else if (language == "rPlot") {
-
-        var url = stringify(this.generateQuery(this.props.query));
-        url = window.location.origin + "/api/v1.0/generatescript/" + url + "/r/" + "PLOT/";
-      } else {
-
-        var url = stringify(this.generateQuery(this.props.query));
-        if (language == "pythonCSV") {
-          url = window.location.origin + "/api/v1.0/generatescript/" + url + "/python/" + "CSV/";
-        } else if (language == "rCSV") {
-          url = window.location.origin + "/api/v1.0/generatescript/" + url +
-          `&save&format=csv&size=${this.props.query.size}` +
-          `&dpi=${this.props.query.dpi}` + "/r/" + "CSV/";
-        }
-      
+        var url = window.location.origin + "/api/v1.0/generatescript/?query=" +query+ "&lang=r" + "&scriptType=PLOT";
+      } else if (language == "pythonCSV") {
+        var url = window.location.origin + "/api/v1.0/generatescript/?query=" + query + "&lang=python" +"&scriptType=CSV";
+      } else if (language == "rCSV") {
+        var url = window.location.origin + "/api/v1.0/generatescript/?query=" + query+ "&lang=r" + "&scriptType=CSV";
```

Comment:
Javascript support `switch` statements on strings so we could actually simplify this a bit further to reduce code duplication. For example:

```js
const query = encodeURIComponent(stringify(this.generateQuery(this.props.query)));

let scriptLang = null;
let scriptType = null;
switch (language) {
  case "pythonPlot" {
    scriptLang = "python";
    scriptType = "PLOT";
    break;
  }
  // Add other cases for rPlot, pythonCSV, and rCSV!


  default:
    console.error(`Unknown language {language}.`);
    break;
}

const url = `${window.location.origin}/api/v1.0/generatescript/?query=${query}&lang=${scriptLang}&scriptType=${scriptType}`;

```

Notice my use of back-ticks (`) to use string interpolation syntax: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
As a side-note we avoid declaring new variables in Javascript using `var` since that's basically a global variable which is bad. https://chris-franklin.com/why-is-var-considered-a-bad-practice-in-javascript/

Reply:
Yes good one, switch case will be a perfect fit for this task. And, yes the variable needs to be declared as a _const_. Making the changes now.